compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1223,4 +1224,26 @@ object messages {
            |Attempting to define a field in a method signature after a varargs field is an error.
            |""".stripMargin
   }
+
+  case class AmbiguousOverload(tree: tpd.Tree, alts: List[SingleDenotation], pt: Type)(
+    err: typer.ErrorReporting.Errors)(
+    implicit ctx: Context)
+  extends Message(AmbiguousOverloadID) {
+
+    private def all = if (alts.length == 2) "both" else "all"
+
+    override def msg: String =
+      s"""|Ambiguous overload. The ${err.overloadedAltsStr(alts)}
+          |$all match ${err.expectedTypeStr(pt)}""".stripMargin
+
+    override def kind: String = "Reference"
+
+    override def explanation: String =
+       s"""|There are ${alts.length} methods that could be referenced as the compiler knows too little about the expected type.
```

Comment:
Yeah, you're completely right. There are some cases where we don't want it to wrap. For instance on lines beginning with `-, +, *,  , ` and so on... maybe we could just principle it on wrapping over 80 characters until it hits two newlines or one of the disallowed symbols. It would also need to take into account ANSI escape codes for the colors.